### PR TITLE
Improving coverage display? #285 => Add GutterFormatter

### PR DESCRIPTION
**Pull request by Guymestef**
Here a first implementation using gutters

### CODE REVIEW

**Comment by orta** [comment]
Hi sorry for the late reply, any chance we can get a screenshot or two to see what this looks like?

**Comment by Guymestef** [comment]
Hi, here a screenshot
sans titre

**Comment by Guymestef** [comment]
For info: green for covered, red for not covered, orange for partially covered branch.
Btw, on some lines the coverage looks weird. I didn't investigate since the displayed coverage gives the same with both the default formater and the gutter formater. But I'm wondering if there isn't a problem with the loading of the coverage file...

**Comment by orta** [comment]
Ah, damn, I forgot to come back and merge - IMO this looks good. 

Any chance I can get this rebased, and we can get it in?

**Comment by Guymestef** [comment]
You mean rebasing my 2 commits onto vscode-jest master or merging vscode-jest master into my master and resolving conflicts?

**Comment by orta** [comment]
Either is fine 👍

**Comment by orta** [comment]
Nice one - thanks, I'll submit a new extension release after another PR has come through

File: package.json [hunk 1]
```diff
@@ -89,6 +89,12 @@
           "type": "boolean",
           "default": false
         },
+        "jest.coverageFormatter": {
+          "description": "Coverage formatter to use",
+          "type": "string",
+          "enum": ["DefaultFormatter", "GutterFormatter"],
+          "default": "DefaultFormatter"
+        },
```
Comment (seanpoulter) [inline package.json]: I <3 the `enum`.  👍

File: src/Coverage/CoverageOverlay.ts [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
-    this.formatter = new DefaultFormatter(coverageMapProvider)
+    switch (coverageFormatter) {
+      case 'GutterFormatter':
+        this.formatter = new GutterFormatter(context, coverageMapProvider)
+        break
+
+      default:
+        this.formatter = new DefaultFormatter(coverageMapProvider)
+        break
+    }
```
Comment (seanpoulter) [inline src/Coverage/CoverageOverlay.ts]: 👍 👍 👍

**Pull request merged**